src/storage/v2/storage.cpp

```diff
@@ -1962,4 +1995,38 @@ void Storage::SetIsolationLevel(IsolationLevel isolation_level) {
   isolation_level_ = isolation_level;
 }
 
+void Storage::RestoreReplicas() {
+  MG_ASSERT(memgraph::storage::ReplicationRole::MAIN == GetReplicationRole());
```

Comment:
I'm not sure if we want to support it, but it can be operationally legitimate to make one replica be the main for another replica in a chain. I think it's fine to assert this though and explicitly not support chain replication, because that causes a lot of additional correctness considerations (what happens if a replica is sync, and that replica has a sync replica, and they basically have to replicate all the way down the chain before main ever gets a success response on the happy path).

Reply:
That can make sense in the new architecture. I wouldn't change it here though, since we have the same checks for registering/"un-registering" replicas:
https://github.com/memgraph/memgraph/blob/b737e53456e0bfb225ff9967b5379f9d3927beb8/src/storage/v2/storage.cpp#L1883-L1886